cmd/cosign/cli/attach/sbom.go

```diff
@@ -0,0 +1,101 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package attach
+
+import (
+	"context"
+	"flag"
+	"io/ioutil"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	"github.com/peterbourgon/ff/v3/ffcli"
+
+	"github.com/sigstore/cosign/pkg/cosign"
+	cremote "github.com/sigstore/cosign/pkg/cosign/remote"
+)
+
+func SBOM() *ffcli.Command {
+	var (
+		flagset = flag.NewFlagSet("cosign attach sbom", flag.ExitOnError)
+		sbom    = flagset.String("sbom", "", "path to the sbom, or {-} for stdin")
+	)
+	return &ffcli.Command{
+		Name:       "sbom",
+		ShortUsage: "cosign attach sbom <image uri>",
+		ShortHelp:  "attach sbom to the supplied container image",
+		FlagSet:    flagset,
+		Exec: func(ctx context.Context, args []string) error {
+			if len(args) != 1 {
+				return flag.ErrHelp
+			}
+
+			return SBOMCmd(ctx, *sbom, args[0])
+		},
+	}
+}
+
+func SBOMCmd(ctx context.Context, sbomRef, imageRef string) error {
+
+	ref, err := name.ParseReference(imageRef)
+	if err != nil {
+		return err
+	}
+
+	b, err := ioutil.ReadFile(sbomRef)
+	if err != nil {
+		return err
+	}
+	s := &cremote.StaticLayer{
+		B: b,
+		// TODO: Fix me.
+		Mt: types.MediaType("application/vnd.sbom.content.layer.v1+sbom"),
+	}
+
+	img := mutate.MediaType(empty.Image, types.OCIManifestSchema1)
+	img, err = mutate.Append(img, mutate.Addendum{
+		Layer: s,
+	})
+	if err != nil {
+		return err
+	}
+
+	m, err := img.Manifest()
+	if err != nil {
+		return err
+	}
+	// TODO: Fix me.
+	m.Config.MediaType = "application/vnd.sbom.config.v1+json"
```

Comment:
@jonjohnsonjr what do i put here

Reply:
Is there any useful configuration you'd want to put in there?

Reply:
No. I could eventually see the need for multiple SBOMs (one for the overall image, then some for sub-parts), but that would get tackled via annotations on each layer to indicate which portion of the image the SBOM is intended to cover.